src/algebra/group/units.lean

```diff
@@ -290,6 +291,10 @@ by cases ha with a ha; rw [←ha, units.mul_left_inj]
 noncomputable def is_unit.unit [monoid M] {a : M} (h : is_unit a) : units M :=
 classical.some h
 
+@[to_additive]
+noncomputable def is_unit.equiv (M : Type*) [monoid M] : equiv {a : M | is_unit a} (units M) :=
+equiv.symm $ equiv.of_injective _ units.ext
```

Comment:
I think this might be better stated as
```
@[to_additive]
noncomputable def is_unit.subtype_equiv (M : Type*) [monoid M] : subtype (is_unit : M → Prop) ≃ (units M) :=
equiv.symm $ equiv.of_injective _ units.ext
```
since your current statement is really saying `{x : M // x ∈ {x : M | is_unit x}} ≃ (units M)`

Reply:
I don't really understand this either: why `{x : M // x ∈ {x : M | is_unit x}}` is more explicative and why your statement is clearer?